src/main/java/com/untamedears/ItemExchange/utility/BlockUtility.java

```diff
@@ -0,0 +1,111 @@
+package com.untamedears.ItemExchange.utility;
+
+import org.bukkit.Bukkit;
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.block.*;
+import org.bukkit.inventory.*;
+
+import com.untamedears.ItemExchange.DeprecatedMethods;
+import com.untamedears.ItemExchange.ItemExchangePlugin;
+
+public class BlockUtility {
+
+	public static final BlockFace[] CardinalFaces = new BlockFace[] {
+			BlockFace.NORTH,
+			BlockFace.EAST,
+			BlockFace.SOUTH,
+			BlockFace.WEST,
+			BlockFace.DOWN,
+			BlockFace.UP
+	};
+
+	public static BlockFace GetFacingDirection(Block block) {
+		byte data = DeprecatedMethods.getBlockMeta(block);
+		if ((data & 0x5) == 5)
+			return BlockFace.EAST;
+		else if((data & 0x4) == 4)
+			return BlockFace.WEST;
+		else if((data & 0x3) == 3)
+			return BlockFace.SOUTH;
+		else if((data & 0x2) == 2)
+			return BlockFace.NORTH;
+		else if((data & 0x1) == 1)
+			return BlockFace.UP;
+		else if((data & 0x0) == 0)
+			return BlockFace.DOWN;
+		else
+			return BlockFace.SELF;
+	}
+
+	public static BlockFace GetAttachedDirection(Block block) {
+		byte data = DeprecatedMethods.getBlockMeta(block);
+		if ((data & 0x5) == 5)
+			return BlockFace.UP;
+		else if((data & 0x4) == 4)
+			return BlockFace.NORTH;
+		else if((data & 0x3) == 3)
+			return BlockFace.SOUTH;
+		else if((data & 0x2) == 2)
+			return BlockFace.WEST;
+		else if((data & 0x1) == 1)
+			return BlockFace.EAST;
+		else if((data & 0x0) == 0)
+			return BlockFace.DOWN;
+		else
+			return BlockFace.SELF;
+	}
+
+	public static void PowerBlock(Block block, long time) {
+		// Ensure that timedelay exists
+		if (time < 0) time = 0;
+		// Set the block to be powered
+		BlockState B_OnState = block.getState();
+		byte B_OnMeta = DeprecatedMethods.getBlockMeta(block);
+		DeprecatedMethods.setBlockMeta(B_OnState, (byte)(B_OnMeta | 0x8));
+		B_OnState.update();
+		// And set the block to unpower in due time
+		Bukkit.getScheduler().scheduleSyncDelayedTask(ItemExchangePlugin.instance, new Runnable() {
+			public void run() {
+				BlockState B_OffState = block.getState();
+				byte B_OffMeta = DeprecatedMethods.getBlockMeta(block);
+				DeprecatedMethods.setBlockMeta(B_OffState, (byte)(B_OffMeta & ~0x8));
+				B_OffState.update();
+			}
+		}, time);
+	}
+
+	public static boolean IsShopCompatible(Material mat) {
```

Comment:
Method name should start with lower case

Reply:
Also this method is rather unneccesary, you could just call `ItemExchangePlugin.ACCEPTABLE_BLOCKS.contains(mat)`, because ACCEPTABLE_BLOCKS is a list

Reply:
1. Done.

2. Done. (I made that function originally because I wasn't sure if .contains() uses == or !=, or whatever mechanism, because the types don't support !=, which is why you have to do !(something == other), and then kept it because IsShopCompatible() was more readable.)